### PR TITLE
feat: add create retrieve and update api endpoints for organization resource

### PR DESCRIPTION
**Pull request by hrithikesh026**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Added `/organization` route to do crud operation on organization table.

Authentication method for `/organization` will be `admin-api-key`

Also generated openapi spec for the new api models and endpoints.

New Routes:
create: POST->` {base_url}/organization/`
retrieve: Get-> `{base_url}/organization/:id`
update: Put-> `{base_url}/organization/:id`

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?

Manual
1. create an organization
```
curl --location 'http://localhost:8080/organization' \
--header 'Content-Type: application/json' \
--header 'api-key: test_admin' \
--data '{
    "organization_name": "my_org"
}'
```

Response:
```
{
    "organization_id": "org_q98uSGAYbjEwqs0mJwnz",
    "organization_name": "my_org"
}
```

2. Retrieve the organization
```
curl --location --request GET 'http://localhost:8080/organization/org_q98uSGAYbjEwqs0mJwnz' \
--header 'Content-Type: application/json' \
--header 'api-key: test_admin' \
--data '{
    "organization_name": "my_org"
}'
```
Response:
```
{
    "organization_id": "org_q98uSGAYbjEwqs0mJwnz",
    "organization_name": "my_org"
}
```

3. Update the organization
```
curl --location --request PUT 'http://localhost:8080/organization/org_q98uSGAYbjEwqs0mJwnz' \
--header 'Content-Type: application/json' \
--header 'api-key: test_admin' \
--data '{
    "organization_name": "your_org"
}'
```
Response:
```
{
    "organization_id": "org_q98uSGAYbjEwqs0mJwnz",
    "organization_name": "your_org"
}
```

DB screenshot.
image

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: crates/router/src/core/admin.rs [hunk 1]
```diff
@@ -106,6 +106,69 @@ fn add_publishable_key_to_decision_service(
...
+        .map(ForeignFrom::foreign_from)
+        .map(service_api::ApplicationResponse::Json)
+}
+
+#[cfg(feature = "olap")]
+pub async fn get_organization(
+    state: SessionState,
+    org_id: api::OrganizationId,
+) -> RouterResponse<api::OrganizationResponse> {
+    #[cfg(all(not(feature = "v2"), feature = "olap"))]
```
Comment (jarnura) [inline crates/router/src/core/admin.rs:154]: use v1 and v2 conventions

**Review by jarnura — CHANGES_REQUESTED** [review]

File: crates/router/src/core/admin.rs (hunk 1 above)
Comment (hrithikesh026) [reply crates/router/src/core/admin.rs:153]: This change will be made once this pr gets merged.
https://github.com/juspay/hyperswitch/pull/5365

File: crates/api_models/src/admin.rs [hunk 2]
```diff
@@ -101,7 +101,8 @@ pub struct MerchantAccountCreate {
     pub frm_routing_algorithm: Option<serde_json::Value>,
 
     /// The id of the organization to which the merchant belongs to, if not passed an organization is created
-    pub organization_id: Option<String>,
+    #[schema(value_type = String, max_length = 64, min_length = 1, example = "org_q98uSGAYbjEwqs0mJwnz")]
```
Comment (Narayanbhat166) [inline crates/api_models/src/admin.rs:104]: this should be optional

File: crates/common_utils/src/id_type.rs [hunk 3]
```diff
@@ -168,11 +167,11 @@ where
...
-impl<const MAX_LENGTH: u8, const MIN_LENGTH: u8> Display for LengthId<MAX_LENGTH, MIN_LENGTH> {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        write!(f, "{}", self.0 .0)
-    }
-}
+// impl<const MAX_LENGTH: u8, const MIN_LENGTH: u8> Display for LengthId<MAX_LENGTH, MIN_LENGTH> {
+//     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+//         write!(f, "{}", self.0. 0)
+//     }
+// }
```
Comment (sahkal) [inline crates/common_utils/src/id_type.rs:174]: can u remove this comment

File: crates/hyperswitch_domain_models/src/payments/payment_attempt.rs [hunk 4]
```diff
@@ -540,16 +544,24 @@ impl behaviour::Conversion for PaymentIntent {
...
         key: &masking::Secret<Vec<u8>>,
+        key_store_ref_id: String,
     ) -> CustomResult<Self, ValidationError>
     where
         Self: Sized,
     {
         async {
-            let inner_decrypt = |inner| decrypt(inner, key.peek());
+            let inner_decrypt = |inner| {
+                decrypt(
```
Comment (sahkal) [inline crates/hyperswitch_domain_models/src/payments/payment_attempt.rs:558]: I don't think we have decrypt imported anywhere

**Review by sahkal — DISMISSED** [review]

File: crates/common_utils/src/id_type.rs (hunk 3 above)
Comment (hrithikesh026) [reply crates/common_utils/src/id_type.rs:174]: Will be done in this PR. https://github.com/juspay/hyperswitch/pull/5380

File: crates/hyperswitch_domain_models/src/payments/payment_attempt.rs (hunk 4 above)
Comment (hrithikesh026) [reply crates/hyperswitch_domain_models/src/payments/payment_attempt.rs:558]: will be taken up in this PR.
https://github.com/juspay/hyperswitch/pull/5380

**Review by lsampras — APPROVED** [review]

**Review by Narayanbhat166 — APPROVED** [review]

File: crates/hyperswitch_domain_models/src/payments/payment_attempt.rs [hunk 5]
```diff
@@ -1,10 +1,14 @@
 use api_models::enums::Connector;
 use common_enums as storage_enums;
+#[cfg(all(any(feature = "v1", feature = "v2"), not(feature = "payment_v2")))]
+use common_utils::types::keymanager::KeyManagerState;
+#[cfg(all(feature = "v2", feature = "payment_v2"))]
+use common_utils::types::keymanager::KeyManagerState;
```
[the remark this answers is not in the document]
Comment (hrithikesh026) [reply crates/hyperswitch_domain_models/src/payments/payment_attempt.rs:6]: cargo clippy --no-default-features will say unused imports.

File: crates/common_utils/src/id_type/organization.rs [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+    pub(crate) fn new(
+        organization_id: LengthId<
+            MAX_ALLOWED_MERCHANT_REFERENCE_ID_LENGTH,
+            MIN_REQUIRED_MERCHANT_REFERENCE_ID_LENGTH,
+        >,
+    ) -> Self {
+        Self(organization_id)
+    }
+
+    /// Get the string representation of customer id
```
[the remark this answers is not in the document]
Comment (hrithikesh026) [reply crates/common_utils/src/id_type/organization.rs:63]: Will be done in this PR: https://github.com/juspay/hyperswitch/pull/5380

**Review by lsampras — APPROVED** [review]

**Pull request merged**